src/main/java/org/folio/rest/impl/EholdingsTagsImpl.java

```diff
@@ -0,0 +1,102 @@
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Context;
+import io.vertx.core.Handler;
+import io.vertx.core.Vertx;
+import io.vertx.core.logging.Logger;
+import io.vertx.core.logging.LoggerFactory;
+import org.apache.commons.collections4.CollectionUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.convert.converter.Converter;
+
+import org.folio.rest.annotations.Validate;
+import org.folio.rest.aspect.HandleValidationErrors;
+import org.folio.rest.jaxrs.model.TagCollection;
+import org.folio.rest.jaxrs.resource.EholdingsTags;
+import org.folio.rest.tools.utils.TenantTool;
+import org.folio.rest.util.ErrorHandler;
+import org.folio.rest.util.ErrorUtil;
+import org.folio.rest.validator.RectypeParameterValidator;
+import org.folio.spring.SpringContextUtil;
+import org.folio.tag.Tag;
+import org.folio.tag.repository.TagRepository;
+
+public class EholdingsTagsImpl implements EholdingsTags {
+
+  private final Logger log = LoggerFactory.getLogger(EholdingsTagsImpl.class);
+
+  @Autowired
+  private RectypeParameterValidator recordTypeValidator;
+  @Autowired
+  private TagRepository tagRepository;
+  @Autowired
+  private Converter<List<String>, Set<org.folio.tag.RecordType>> recordTypesConverter;
+  @Autowired
+  private Converter<List<Tag>, TagCollection> tagsConverter;
+
+
+  public EholdingsTagsImpl() {
+    SpringContextUtil.autowireDependencies(this, Vertx.currentContext());
+  }
+
+  @Validate
+  @HandleValidationErrors
+  @Override
+  public void getEholdingsTags(List<String> filterRectypes, Map<String, String> okapiHeaders,
+      Handler<AsyncResult<Response>> asyncResultHandler, Context vertxContext) {
+    validateRecordTypes(filterRectypes);
+
+    completedFuture(null)
+      .thenCompose(o -> findTags(filterRectypes, TenantTool.tenantId(okapiHeaders)))
+      .thenAccept(tags -> successfulGetTags(tags, asyncResultHandler))
+      .exceptionally(throwable -> failedGetTags(throwable, asyncResultHandler));
+  }
+
+  private void validateRecordTypes(List<String> filterRectypes) {
+    if (CollectionUtils.isNotEmpty(filterRectypes)) {
+      filterRectypes.forEach(recordTypeValidator::validate);
+    }
+  }
+
+  private CompletableFuture<List<Tag>> findTags(List<String> filterRectypes, String tenantId) {
+    log.info("Retrieving tags: tenantId = {}, recordTypes = {}", tenantId, filterRectypes);
+
+    if (CollectionUtils.isEmpty(filterRectypes)) {
+      return tagRepository.findAll(tenantId);
+    } else {
+      return tagRepository.findByRecordTypes(tenantId, recordTypesConverter.convert(filterRectypes));
+    }
+  }
+
+  private void successfulGetTags(List<Tag> tags, Handler<AsyncResult<Response>> handler) {
+    handler.handle(succeededFuture(GetEholdingsTagsResponse.respond200WithApplicationVndApiJson(
+      tagsConverter.convert(tags))));
+  }
+
+  private Void failedGetTags(Throwable th, Handler<AsyncResult<Response>> handler) {
```

Comment:
void?

Reply:
method is used in lambda. it's return type has to be and object, in this case it's Void

Reply:
👍